src/blob/index/core.rs

```diff
@@ -344,10 +377,17 @@ where
         }
     }
 
-    fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<u64>> {
+    fn mark_all_as_deleted(&mut self, key: &K, header: RecordHeader) -> Result<u64> {
```

Comment:
`push_deletion`?

Reply:
As I see, this function becomes identical to `push`, which is good. So it is better to call `push` here, but additionally check, that passed `header` has `deleted` flag and its record size is zero